File: 2022-2023/sourcecode/lab/lab7/NotyJava/app/src/main/java/pl/udu/uwr/pum/notyjava/adapters/NotyAdapter.java
package pl.udu.uwr.pum.notyjava.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import pl.udu.uwr.pum.notyjava.databinding.RecyclerItemViewBinding;
import pl.udu.uwr.pum.notyjava.fragments.NotesFragmentDirections;
import pl.udu.uwr.pum.notyjava.model.NoteModel;

public class NotyAdapter extends RecyclerView.Adapter<NotyAdapter.ViewHolder> {

    private final ArrayList<NoteModel> notes;

    public NotyAdapter(ArrayList<NoteModel> notes){
        this.notes = notes;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(RecyclerItemViewBinding.inflate(LayoutInflater.from(
                parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        NoteModel item = notes.get(position);
        holder.bind(item);
        holder.binding.getRoot().setOnClickListener(v -> {
            NavDirections action = NotesFragmentDirections
                .actionNotesFragmentToEditNoteFragment(item.getId());
            Navigation.findNavController(holder.binding.getRoot()).navigate(action);
        });
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final RecyclerItemViewBinding binding;

        public ViewHolder(RecyclerItemViewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
        public void bind(NoteModel item){
            binding.timeTextView.setText(item.getTime().toString());
            binding.textTextView.setText(item.getTextNote());
            binding.timeTextView.setTextColor(item.getColor());
            binding.textTextView.setTextColor(item.getColor());
        }
    }
}

File: 2022-2023/sourcecode/lab/lab7/NotyKotlin/app/src/main/java/pl/udu/uwr/pum/notykotlin/adapters/NotyAdapter.kt
package pl.udu.uwr.pum.notykotlin.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.navigation.NavDirections
import androidx.navigation.Navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import pl.udu.uwr.pum.notykotlin.databinding.RecyclerItemViewBinding
import pl.udu.uwr.pum.notykotlin.fragments.NotesFragmentDirections
import pl.udu.uwr.pum.notykotlin.model.NoteModel

class NotyAdapter(private val notes: List<NoteModel>) :
    RecyclerView.Adapter<NotyAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            RecyclerItemViewBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item: NoteModel = notes[position]
        holder.bind(item)
        holder.binding.root.setOnClickListener {
            val action: NavDirections = NotesFragmentDirections
                .actionNotesFragmentToEditNoteFragment(item.id)
            findNavController(holder.binding.root).navigate(action)
        }
    }

    override fun getItemCount(): Int {
        return notes.size
    }

    class ViewHolder(val binding: RecyclerItemViewBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun bind(item: NoteModel) {
            binding.timeTextView.text = item.time.toString()
            binding.textTextView.text = item.textNote
            binding.timeTextView.setTextColor(item.color)
            binding.textTextView.setTextColor(item.color)
        }
    }
}